app/controllers/carts_controller.rb

```diff
@@ -8,14 +18,20 @@ def checkout_cart
     end
   end
 
-  def create
-    update_cart
-    flash[:notice] = 'Added to Cart'
-    redirect_to items_path
+  def delete_item
+    if params[:cart].nil?
+      delete_all_items_from_cart
+    else
+      delete_specific_item_from_cart
+      flash[:notice] = 'Item removed from cart'
+    end
+    redirect_to showcart_path
   end
 
-  def showcart
-    @cart = Cart.new(session[:cart]) if session[:cart]
+  def update_item_quantity
+    session[:cart][params[:update_item_quantity][:item_id]] = params[:update_item_quantity][:quantity]
+    flash[:notice] = 'Item quantity updated'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.